docs/adr/0003-delivery-channels-templating.md

```diff
@@ -0,0 +1,92 @@
+# Templating engine for different notification types
+
+## Status
+
+Proposed
+
+## Context
+
+This ADR proposes a method on how to implement custom templating when a certain notification is sent through a communication channels (SMS, email, web push notifications etc).
+
+In a typical notification system, an object will be applied to a basic template to provide an end user with information targeting them. A lot of the basic design/layout of the template will be quite standard, but there will be places for values that will need to be inserted to customise it for the said end user. An example of this would be an birthday email notification, where a lot of the design and formatting of the email/subject will already be very well defined, and then insertion of custom data (eg. email id, name, etc ) is what will be needed by a templater.
+
+The content of this ADR will work off themes from the previous ADR (0002-delivery-channel-creation-and-integration.md).
+
+## Decision
+
+A notification having necessary information needs to be translated into a certain communication object (SMS, Email, etc) before it can be sent via the desired delivery channel (SendGrid, Twillio, Airship).
+
+An example of this would be a SMS, which needs the essential information to be available for it to be converted into an SMS object before being sent to the end user.
+
+```json
+
+    public class SMS {
+        public string ToPhoneNumber { get; set; }
+        public string Body { get; set; }
+    }
+
+```
+
+Each field in the communication object will needs it's own template (property) for the associated notification type. A template for the `ToPhoneNumber` property in the above example would look something like:
+
+```
+{{ notification.propertyBag.ToPhoneNumber }}
```

Comment:
This makes it look like the `ToPhoneNumber` is attached to the notification itself. Is that the intent? That would suggest that the code which raises the notification will need to attach that phone number to the data it sends through. And assuming we don't want every point in client code that creates a notification to need to know which channels the notification will be sent via, then that means the client code would need to attach every possible bit of config for each delivery channel the notification *might* be sent through.

Reply:
Agreed. We would now be storing all the config related to the end user in table storage called user preferences. I have updated the ADR.